pkg/plugins/runtime/universal/outbound/outbound.go

```diff
@@ -63,6 +65,21 @@ func (v *VIPOutboundsReconciler) UpdateVIPOutbounds(ctx context.Context) error {
 	if err := v.rorm.List(ctx, meshes); err != nil {
 		return err
 	}
+
+	// First get all ingresses
+	var ingresses []*mesh.DataplaneResource
+	for _, m := range meshes.Items {
```

Comment:
Thank you a lot for fixing this as well! You are absolutely right about scaling, we have this problem right now, ideally, we need to untie Ingress from the Dataplane model. Probably creating a separate resource for the Ingress is a better approach. 

Just wanted to suggest a small improvement to your code, if you call `v.rorm.List(ctx, dpList)` without passing the mesh explicitly, it will return dataplanes from all the meshes and then you can just filter ingresses

Reply:
The discussion about the separate ingress resource was also raised here:
https://kuma-mesh.slack.com/archives/CN2GN4HE1/p1611583155036000

Reply:
> Just wanted to suggest a small improvement to your code, if you call v.rorm.List(ctx, dpList) without passing the mesh explicitly, it will return dataplanes from all the meshes and then you can just filter ingresses

I thought that pulling dataplanes by mesh would avoid loading all dataplanes at once. Especially as we already have the list of meshes.

I'll update if you think it's simpler this way.